airbyte-cdk/bulk/toolkits/extract-cdc/src/main/kotlin/io.airbyte.cdk/read/CdcPartitionCreator.kt

```diff
@@ -26,6 +26,11 @@ class CdcPartitionCreator(
         return PartitionsCreator.TryAcquireResourcesStatus.READY_TO_RUN
     }
     override suspend fun run(): List<PartitionReader> {
+        // successful acquisition means that CDC is complete
+        if (cdcContext.cdcGlobalLockResource.tryAcquire() != null) {
+            // For the time being, only do one round
+            return listOf()
+        }
```

Comment:
Resource acquisition should happen in `tryAcquireResources`, not in `run`.

Reply:
I agree it's a little weird but I think that the name of the method is a little bit of a misnomer, since nothing is acquired.
Let me try to see how I can improve.

Reply:
also we don't acquire anything in `CdcPartitionReader` and also, if for whatever reason another `CdcPartitionCreator` tried to run while the first cdc sync is running there's no way of stopped it as there's is not distinction between running and already ran.
I gotta say, the original implementation using a LimitedLockMutex had all this.

Reply:
Yeah, you're right about the acquisition. Let's just merge this as-is, it's fine for the time being. I'm going to publish a PR today which is going to change this anyway because it implements multiple rounds. Problem solved!